src/utils.ts

```diff
@@ -0,0 +1,41 @@
+import { DataViewerProposal } from './interfaces/DataViewerProposal';
+
+/**
+   * Safely gets the proposal value for a given proposal attribute key.
+   *
+   * @param  {DataViewerProposal} proposal
+   * @param  {string} key The key for the proposal attribute we want.
+   * @return {(Array|undefined)}
+   */
+const getProposalValue = (proposal: DataViewerProposal, key: string) => {
+  // This is a temporary fix:
+  // The API currently allows fields to be set to blank strings,
+  // and dutifully includes those in the value array.
```

Comment:
This is helpful, but I feel like this is the wrong place to solve this; instead, I think we should be filtering such empty values during mapping the API response into our own data structures.

How comfortable do you feel tackling that? I know this function is a critical component of this work, so I don't want to block the rest of it on doing temporary API cleanup in the "right" place!

Reply:
Oof, I think you're right… although (I hate to say it) maybe after the demo?

Do you think this would belong somewhere like [between L77 and L78](https://github.com/PhilanthropyDataCommons/data-viewer/blob/ac46a38cc9c9fac8e89bbb4826f4983f9fe4a23e/src/pdc-api.ts#L77-L78)?

I wonder what role this utility would even play, after that. Like I mentioned, the shape of this function changed quite a lot as I had tests that let me strip out internals. I'll probably be able to remove it entirely.

Reply:
Okay, I've played around with an implementation of this, where I modified `mapProposals` thusly:

```diff
-        .filter((pair): pair is [string, string] => !!pair[0])
+        .filter((pair): pair is [string, string] => (!!pair[0] && pair[1]?.trim() !== ''))
```

Is this the starting point you were envisioning, @jasonaowen? If so, I can do that as the first commit on this PR, and then modify the following changes accordingly. For instance, we lose the need to even have `getProposalValues` at all if we're guaranteeing there are no blank values.

One consequence is that we'll need to create tests for `mapProposals` to cover some of the scenarios that were being tested by `getProposalValues` and `getProposalValuesFromCandidates` tests, since [structures like this](https://github.com/PhilanthropyDataCommons/data-viewer/pull/101/files#diff-04abd0e283864d1e858d70a5f86f67ccf45fcd3b752e56b136c3eee449b06b28R18-R25) should no longer even exist. And I'd really love to have an integration test that made sure an entire "API response → `mapProposals()` → `getProposalValuesFromCandidates()`" flow worked properly, but that's probably out of scope.

Reply:
Awesome! Yes, that's roughly what I'd expect. I'd suggest adding a subsequent filter, though, since it's for a different purpose - the current filter is mostly a type guard that shouldn't ever filter anything out in practice, whereas the new filter is actually excluding data. Maybe something like:

```diff
         .filter((pair): pair is [string, string] => !!pair[0])
+        .filter(([key, value]) => (value.trim() !== ''))
```

> For instance, we lose the need to even have `getProposalValues` at all if we're guaranteeing there are no blank values.

Yes! That's the goal.

> One consequence is that we'll need to create tests for `mapProposals`...

The middle commit of #106 extracts `mapProposals` to its own file, which should make writing tests for it simpler; I considered doing so as part of that extraction, but decided that was a distraction that we could follow up on later.

Reply:
Somehow I forgot that this was back on my plate! 🤦 On it…

Reply:
@jasonaowen Okay! Here's what I decided to do:

1. Split off the `mapProposals()` modification (plus some tests for it) to #112 [DONE]
2. Add `getProposalValuesFromCandidates()` directly on #102 (and rebase it, etc.) [TODO]

At which point I'll close this PR.

Reply:
After way too much churn, I've realized making this change resolves the entire issue without the need for the non-idiomatic syntactic sugar that `getProposalValuesFromCandidates()` provided. Thus, #112 is able to replace _both_ this PR and #102. 🎉